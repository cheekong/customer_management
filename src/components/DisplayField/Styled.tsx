import styled from 'styled-components';

export const StyledSpan = styled.span `
    width: 100%;
    border: 1px solid lightgray;
    height: 60px;
    margin-top:10px;
    margin-bottom:10px;
    display: flex;
`

export const StyledLabel = styled.label `
    width: 30%;
    padding: 0;
    margin: 0;
    margin-top:auto;
    margin-bottom: auto;
    font-size: 1em;
`

export const StyledParagraph = styled.p `
    width: 70%;
    height:100%;
    padding: 0;
    margin: 0;
    margin-top:auto;
    margin-bottom: auto;
    line-height:60px
    font-size: 1em;
    text-indent:  10px;
    background-color: #f5f5f5;
`