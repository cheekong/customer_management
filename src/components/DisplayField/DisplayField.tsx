import React from 'react';
import {StyledSpan, StyledLabel, StyledParagraph} from './Styled'

interface MyProps {
  label: string;
  pararapgh: string
}

const CustomerList: React.SFC<MyProps> = (props) => {
    return (
      <StyledSpan>
        <StyledLabel>
          {props.label}
        </StyledLabel>
        <StyledParagraph>
          {props.pararapgh}
        </StyledParagraph>
      </StyledSpan>
    );
  }
  
  export default CustomerList;
  