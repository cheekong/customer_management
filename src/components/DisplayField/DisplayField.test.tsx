import React from 'react';
import DisplayField from './DisplayField';
import renderer from 'react-test-renderer';

describe('Display field', () => {
    test('it works', () => {
    
        const component = renderer.create(
            <DisplayField 
                label='Firstname'
                pararapgh='Jim'
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})


