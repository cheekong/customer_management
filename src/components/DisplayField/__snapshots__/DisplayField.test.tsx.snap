// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display field it works 1`] = `
<span
  className="sc-bdVaJa crMpnZ"
>
  <label
    className="sc-bwzfXH uMthG"
  >
    Firstname
  </label>
  <p
    className="sc-htpNat gbIUNe"
  >
    Jim
  </p>
</span>
`;
